@kiva/kv-analytics/src/scripts.js

```diff
@@ -0,0 +1,252 @@
+				y=n.getElementsByTagName(t)[0];y.parentNode.insertBefore(o,y);
+				g.identify=function(i,v,s){g(l,{uid:i},s);if(v)g(l,v,s)};g.setUserVars=function(v,s){g(l,v,s)};g.event=function(i,v,s){g('event',{n:i,p:v},s)};
+				g.anonymize=function(){g.identify(!!0)};
+				g.shutdown=function(){g("rec",!1)};g.restart=function(){g("rec",!0)};
+				g.log = function(a,b){g("log",[a,b])};
+				g.consent=function(a){g("consent",!arguments.length||a)};
+				g.identifyAccount=function(i,v){o='account';v=v||{};v.acctId=i;g(o,v)};
+				g.clearUserCookie=function(){};
+				g.setVars=function(n, p){g('setVars',[n,p]);};
+				g._w={};y='XMLHttpRequest';g._w[y]=m[y];y='fetch';g._w[y]=m[y];
+				if(m[y])m[y]=function(){return g._w[y].apply(this,arguments)};
+				g._v="1.3.0";
+			})(window,document,window['_fs_namespace'],'script','user');
+		} catch (e) {
+			// noop
+		}
+		/* eslint-enable */
+	};
+
+	// Google Tag Manager snippet
+	const insertGTM = () => {
+		/* eslint-disable */
+		(function(w,d,s,l,i){w[l]=w[l]||[];w[l].push({'gtm.start':
+		new Date().getTime(),event:'gtm.js'});var f=d.getElementsByTagName(s)[0],
+		j=d.createElement(s),dl=l!='dataLayer'?'&l='+l:'';j.async=true;j.src=
+		'//www.googletagmanager.com/gtm.js?id='+i+dl;f.parentNode.insertBefore(j,f);
+		})(window,document,'script','dataLayer',process.env.googleTagmanagerId);
+		/* eslint-enable */
+	};
+
+	// Facebook Pixel Code
+	const insertFB = () => {
+		/* eslint-disable */
+		!function(f,b,e,v,n,t,s) {if(f.fbq)return;n=f.fbq=function(){n.callMethod?
+			n.callMethod.apply(n,arguments):n.queue.push(arguments)};
+			if(!f._fbq)f._fbq=n;n.push=n;n.loaded=!0;n.version='2.0';
+			n.queue=[];t=b.createElement(e);t.async=!0;
+			t.src=v;s=b.getElementsByTagName(e)[0];
+			s.parentNode.insertBefore(t,s)}(window, document,'script',
+			'https://connect.facebook.net/en_US/fbevents.js');
+		fbq('init', process.env.fbPixelId);
+		/* eslint-enable */
+	};
+
+	// Algolia Analytics
+	const insertAlgoliaAnalytics = () => {
+		/* eslint-disable */
+		if (typeof String.prototype.startsWith === 'function') {
+			(function(e,a,t,n,s,i,c){e.AlgoliaAnalyticsObject=s,e.aa=e.aa||function(){
+				(e.aa.queue=e.aa.queue||[]).push(arguments)},i=a.createElement(t),c=a.getElementsByTagName(t)
+				[0],i.async=1,i.src="https://cdn.jsdelivr.net/npm/search-insights@1.2.0",
+				c.parentNode.insertBefore(i,c)})(window,document,"script",0,"aa");
+		}
+		/* eslint-enable */
+	};
+
+	// Always load
+	// PerimeterX snippet
+	if (process.env.enablePerimeterx) {
+		/* eslint-disable */
+		(function(){
+			window._pxAppId = process.env.perimeterxAppId;
+			var p = document.getElementsByTagName('script')[0],
+				s = document.createElement('script');
+			s.async = 1;
+			s.src = `${window.location.origin}/${process.env.perimeterxAppId.substring(2)}/init.js`;
+			p.parentNode.insertBefore(s,p);
+		}());
+		/* eslint-enable */
+	}
+
+	// One Trust Cookie Management
+	if (process.env.oneTrust && process.env.oneTrust.enable) {
```

Comment:
Looking ahead at the Nuxt implementation, their docs show we would likely need to implement the OneTrust include in the nuxt.config `head` section using vue-meta functionality. OneTrust would be the primary head include and would have a callback that fires the method on line 152. All of the other script include functions would need to be declared in a file loaded prior to OneTrust so that OneTrust could access and call them once it's loaded.

How does all the config get injected for each script in the nuxt world...not sure yet.

Ultimately, we will probably be forced to break this file up a bit for the NuxtApp and set these up in a more granular way.

We might just consider re-naming this file as "ui-head-scripts" which can then be imported and injected in UI the same way it is today...

Reply:
Re: config: Nuxt has built-in dotenv support, so I think we should lean on using dotenv for configuration. That could possibly include adding dotenv as a dependency of this package, but we could also say it's the responsibility of users of this package ensure the `process.env` object has all the environment variables it needs.

Re: head script loading: I think renaming this file to "ui-head-scripts" and importing it as appropriate in UI and Nuxt apps make sense. For loading the script in Nuxt, maybe we can add the head script from the module? Something like:
```
module.exports = function nuxtKvAnalytics() {
    ...
    this.options.head.script.push({
        vmid: 'kv-analytics-head-script',
        innerHTML: serialize(headScript);
    });
    this.options.head.__dangerouslyDisableSanitizersByTagID['kv-analytics-head-script'] = ['innerHTML'];
    ...
} 
```
Although we might need to do a little work to ensure the config values from process.env make it in there.

Reply:
That's a neat idea @emuvente !!!